lib/BuildSystem/BuildSystem.cpp

```diff
@@ -1570,8 +1570,8 @@ class ShellCommand : public ExternalCommand {
       return false;
     }
     
-    // Collect the discovered dependencies, if used.
-    if (!depsPaths.empty()) {
+    // Collect the discovered dependencies, if used and command was not cancelled.
+    if (!depsPaths.empty() && !bsci.getDelegate().isCancelled()) {
```

Comment:
hm, good point -- they do return `false` (see <https://github.com/apple/swift-llbuild/blob/master/lib/BuildSystem/LaneBasedExecutionQueue.cpp#L225>), need to investigate further